### PR TITLE
FINERACT-1101 Swagger Codegen build using Gradle

### PR DESCRIPTION
**Pull request by Grandolf49**
## Description
This PR addresses FINERACT-1101 issue. Following were the problems earlier:
1. Library generated by `./gradlew generateSwaggerCode` was required to be build using `mvn package` only. If a user executed `./gradlew build`, the build used to fail saying `Could not determine java version from '11.0.8'.` The issue was with the gradle-wrapper of the project. This is resolved by adding the Mustache Template for the `gradle-wrapper.properties` file.

2. Moreover, `mvn package` also failed due to an error in `maven-javadoc-plugin`. It was using the version `2.10.4` and was failing because of that. Version `3.2.0` worked correctly. This is resolved by adding the Mustache template for the `pom.xml` file

3. To be able to use Mustache Templates, we needed `swagger-codegen-cli` version `3.0.21`. But if we used version `3.0.21` to generate swagger code, it would fail throwing an error `Missing OpenAPI File`. This was because the generated OpenAPI file was invalid. This was preventing to use the Mustache Templates. This is resolved in this [PR](https://github.com/apache/fineract/pull/1270).

Updated the `client.md` file with necessary documentation

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [x] API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm has been updated with details of any API changes.

- [x] Integration tests have been created/updated for verifying the changes made.

- [x] All Integrations tests are passing with the new commits.

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by awasum** [comment]
An Apache RAT test is failing. Make sure you run this branch on your local machine and all builds and tests are passing. https://travis-ci.org/github/apache/fineract/builds/720575703

**Comment by Grandolf49** [comment]
Hi @awasum  I checked the error. It is saying 
```
Unapproved Licenses:
fineract-provider/config/swagger/templates/gradle-wrapper.properties.mustache
fineract-provider/config/swagger/templates/pom.mustache
```

Can you help me resolve this?

**Comment by awasum** [comment]
Ignore the file from rat checks as seen in the build.gradle file.

**Comment by Grandolf49** [comment]
@awasum Thanks for the solution. It worked! I'll update the PR.

File: fineract-provider/config/swagger/config.json.template [hunk 1]
```diff
@@ -5,6 +5,7 @@
    "invokerPackage": "org.apache.fineract.client",
    "modelPackage": "org.apache.fineract.client.models",
    "apiPackage": "org.apache.fineract.client.services",
+   "artifactDescription": "A Java and Retrofit2 based client library for Apache Fineract 1.x Platform",
```
Comment (vorburger) [inline fineract-provider/config/swagger/config.json.template]: I think we usually just call it "Apache Fineract" (VS "Fineract CN"), not "1.x" (I think).

```suggestion
   "artifactDescription": "A Java and Retrofit2 based client library for Apache Fineract Platform",
```

File: docs/developers/swagger/client.md [hunk 2]
```diff
@@ -7,20 +7,40 @@ Apache Fineract supports client code generation using [Swagger Codegen](https://
...
+Make sure you are following the supported templates. Otherwise, the generated code will not build correctly.
+
+- [Java Mustache Templates](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources/Java)
+- [Retrofit2 Mustache Templates](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources/Java/libraries/retrofit2)
+
+If you need to add templates for a specific library, you need to follow the same directory structure as present in the links above.
+
+## Validate OpenAPI Spec File
+
+The `resolve` task in [build.gradle](https://github.com/apache/fineract/blob/217e34ba9e4cf04ad7b82fc3a0f8f8e5f8de8042/fineract-provider/build.gradle#L212) file will generate the OpenAPI Spec File for the project. To make sure Swagger Codegen generates a correct library, it is important for the OpenAPI Spec file to be valid. Use [Swagger OpenAPI Validator](https://validator.swagger.io/) to validate the spec file.
```
Comment (vorburger) [inline docs/developers/swagger/client.md]: the use of a particular commit here was probably an oversight, and you meant to use `develop` here, like in line 29? Actually, I wonder if you couldn't just use a relative path (like `...../fineract-provider/...` you what I mean here?)

```suggestion
The `resolve` task in [build.gradle](https://github.com/apache/fineract/blob/develop/fineract-provider/build.gradle#L212) file will generate the OpenAPI Spec File for the project. To make sure Swagger Codegen generates a correct library, it is important for the OpenAPI Spec file to be valid. Use [Swagger OpenAPI Validator](https://validator.swagger.io/) to validate the spec file.
```

**Review by vorburger — CHANGES_REQUESTED** [review]

**Review by vorburger — APPROVED** [review]

**Pull request merged**